### PR TITLE
add scheduleUnlayout method

### PR DESCRIPTION
**Pull request by zhouyx**
#4193

### CODE REVIEW

File: test/functional/test-resources.js [hunk 1]
```diff
@@ -516,6 +516,113 @@ describe('Resources schedulePreload', () => {
 });
 
 
+describe('Resources scheduleUnlayout', () => {
```
Comment (aghassemi) [inline test/functional/test-resources.js]: You can reuse most of the existing scheduleLayout test code. See https://github.com/ampproject/amphtml/pull/4137/files for how I am sharing code between scheduleResume and schedulePause tests.

File: src/service/resources-impl.js [hunk 2]
```diff
@@ -468,6 +468,20 @@ export class Resources {
   }
 
   /**
+   * Schedules unlayout for specified sub-elements that are children of the
+   * parent element. The parent element can choose to send this signal when
+   * it want to unload resources for its children.
```
Comment (dvoytenko) [inline src/service/resources-impl.js]: Please add `@param` jsdocs

**Comment by dvoytenko** [comment]
LGTM with one request for jsdoc

**Comment by zhouyx** [comment]
@aghassemi I reuse the test for pause/resume make it test for pause/resume/unlayout

**Comment by aghassemi** [comment]
@zhoresh LGTM

File: src/base-element.js [hunk 3]
```diff
@@ -627,6 +627,14 @@ export class BaseElement {
   }
 
   /**
+   * @param {!Element|!Array<!Element>} elements
+   * @protected
+   */
+  scheduleUnlayout(elements) {
```
Comment (dvoytenko) [inline src/base-element.js]: Please add to presubmit restrictions.

**Pull request merged**